pkg/channel/lookup.go

```diff
@@ -45,6 +45,7 @@ func GetTemplates(
 		switch {
 		case module.RemoteModuleTemplateRef == "":
 			template = NewTemplateLookup(kymaClient, module, kyma.Spec.Channel).WithContext(ctx)
+			saveDescriptorToCache(template)
```

Comment:
it's better handle `template.Err != nil` before `saveDescriptorToCache` in case you deal with some err template, also it prevents `template.Err` to be overwritten later in case it has error in  `saveDescriptorToCache`